hw/ip/csrng/doc/_index.md

```diff
@@ -91,3 +91,391 @@ This document aims to make the distinction between physical entropy and factory
+
+
+
+## Compatibility
+None.
+
+# Theory of Operations
+
+The CSRNG block has been constructed to follow the NIST recommendation for a DRBG mechanism based on block ciphers.
+Specifically, it is a CTR_DRBG that uses an approved block cipher algorithm in counter mode.
+As such, the block diagram below makes reference to hardware blocks that either directly or closely follow NIST descriptions for the equivalent functions.
+
+There are two major hardware interfaces: the application interface and the entropy request interface.
+The application interface, which is described in more detail later, is provided for an application to manage an `instance` in CSRNG.
+Once setup, the application interface user can request for entropy bits to be generated, as well as other functions.
+The application interface supports up to 15 hardware interfaces, and one software interface.
+
+A walk through of how CSNRG generates entropy bits begins with the application interface.
+An `instantiate` command is issued from one of the application interfaces.
+This request moves into the `cmd_stage` block.
+Here the request is arbitrated between all of the `cmd_stage` blocks.
+The winner will get its command moved into the command dispatch logic.
+A common state machine will process all application interface commands in order of arbitration.
+At this point, some seed entropy may be required depending on the command and any flags.
+If needed, a request to the entropy source hardware interface will be made.
+This step can take milliseconds if seed entropy is not immediately available.
+Once all of the prerequisites have been collected, a ctr_drbg command can be launched.
+This command will go into the ctr_drbg_cmd block.
+This ctr_drbg_cmd block uses two NIST-defined functions, the update and the block_encrypt functions.
+If the command is a generate, the ctr_drbg_cmd block will process the first half of the algorithm, and then pass it on to the ctr_drg_gen block.
+Additionally, the ctr_drbg_gen block also uses the update block and the block_encrypt block.
+To keep resources to a minimum, both of these blocks have arbiters to allow sharing between the ctr_drbg_cmd and ctr_drbg_gen blocks.
+The command field called `ccmd` (for current command) is sent along the pipeline to not only identify the command, but is also reused as a routing tag for the arbiters to use when returning the block response.
+
+Once the command has traversed through all of the ctr_drbg blocks, the result will eventually land into the state_db block.
+This block will hold the instance state for each application interface.
+The specific state information held in the instance is documented below.
+If the command was a `generate` command, the genbits data word will be returned to the requesting `cmd_stage` block.
+Finally, an `ack` response and status will be returned to the application interface once the command has been completely processed.
+
+
+## Block Diagram
+
+![CSRNG Block Diagram](csrng_blk_diag.svg)
```

Comment:
i'm not quite sure why, but when i view this diagram, i see a few feint grey arrows pointing to nowhere.
Could they just be left over from a previous verion?

Reply:
@rswarbrick has also reported graphics flaws. All of my views, both source and markdown inclusion, do not show any problems, so maybe @martin-lueker can take a look at this.

Reply:
Yes.  For some reason the original Visio diagrams have these.  I can remove them.

Reply:
Phantom Grey-lines removed, and this also has fixed the bounding-box problems.